src/System.Private.ServiceModel/src/System/ServiceModel/Channels/HttpChannelFactory.cs

```diff
@@ -581,9 +503,8 @@ protected override void OnClose(TimeSpan timeout)
 
             protected internal override async Task OnCloseAsync(TimeSpan timeout)
             {
+                var timeoutHelper = new TimeoutHelper(timeout);
                 PrepareClose(false);
-                TimeoutHelper timeoutHelper = new TimeoutHelper(timeout);
```

Comment:
Just checking -- this change is bug fix to make the PrepareClose happen inside the timeout interval, right?